runtime/rasdump/javadump.cpp

```diff
@@ -4619,22 +4676,32 @@ JavaCoreDumpWriter::writeThreadName(J9VMThread* vmThread)
 {
 	PORT_ACCESS_FROM_PORT(_PortLibrary);
 	if (NULL != vmThread) {
-		void *args[] = { _VirtualMachine, vmThread };
-		const char *nameClean = "";
-		const char *nameFault = nameClean;
-
-		/* This can crash if the thread object is being moved around while we're trying to get the
-		 * name out of it (i.e. we're sharing exclusive with GC). If we fault while trying to get the
-		 * name we return "<name unavailable>" instead
-		 */
-		if (j9sig_protect(protectedGetVMThreadName, args, handlerGetVMThreadName, (UDATA*)&nameFault, J9PORT_SIG_FLAG_SIGALLSYNC | J9PORT_SIG_FLAG_MAY_RETURN, (UDATA*)&nameClean) == J9PORT_SIG_EXCEPTION_OCCURRED) {
-			_OutputStream.writeCharacters(nameFault);
-		} else if (NULL != nameClean) {
-			_OutputStream.writeCharacters(nameClean);
-		} else {
-			_OutputStream.writeCharacters("<name locked>");
+#if JAVA_SPEC_VERSION >= 21
+		if ((NULL != vmThread->threadObject) && IS_JAVA_LANG_VIRTUALTHREAD(vmThread, vmThread->threadObject)) {
+			/* For VirtualThread, get name from threadObject directly. */
```

Comment:
I'm not sure this is wise (safe), for the same reason `j9sig_protect()` is used for non-virtual threads.

Reply:
I went through the call chain of `protectedGetVMThreadName`, it is only using the `vmthread->omrVMThread->threadName` field and didn't make use of any java object which shouldn't even need the `j9sig_protect` wrap.

for access to `vmThread->threadObject`, I agree there should be a guard for the case of thread object is being moved around, only issue is the virtual thread check would also need to be wrapped. so I am thinking of moving the checking/output section into `protectedGetVMThreadName` and add a new protected helper to retrieve data from thread object.

`JavaCoreDumpWriter::writeThreadBlockers` also have a use of `getVMThreadNameFromString` which isn't wrapped, should that be fixed as well?

Reply:
@keithc-ca do you have any suggestion for a better approach?

Reply:
I think moving the checks into `protectedGetVMThreadName()` is reasonable. To be clear, I think it should not print anything, just return the name (or some failure string) if it fails.

`writeThreadBlockers()` is only called from `protectedWriteThreadBlockers()`, so a fault within the call to `writeThreadName()` there would only affect the output for that one thread. Ideally, a fault that occurs while printing information about one unmounted thread would not prevent printing information about other, following unmounted threads. So we want to use `j9sig_protect()` judiciously to limit the loss of information due to faults.